include/fmt/core.h

```diff
@@ -1468,13 +1548,38 @@ class dynamic_format_arg_store
   friend class basic_format_args<Context>;
 
   unsigned long long get_types() const {
-    return internal::is_unpacked_bit | data_.size();
+    return internal::is_unpacked_bit | data_.size() |
+           (named_info_.empty() ? 0ULL
+                                : static_cast<unsigned long long>(
+                                      internal::has_named_args_bit));
+  }
+
+  const basic_format_arg<Context>* data() const {
+    return named_info_.empty() ? data_.data() : data_.data() + 1;
   }
 
   template <typename T> void emplace_arg(const T& arg) {
     data_.emplace_back(internal::make_arg<Context>(arg));
   }
 
+  template <typename T>
+  void emplace_arg(const internal::named_arg<T, char_type>& arg) {
+    if (named_info_.empty()) {
+      data_.insert(
+          data_.begin(),
+          {static_cast<const internal::named_arg_info<char_type>*>(nullptr),
```

Comment:
We can avoid cast and construct the pointer directly: `const internal::named_arg_info<char_type>*()`

Reply:
Actually it seems to be impossible initializing anonymous pointer in such a way without adding alias to the type. 
Interestingly cppreference tells nothing about non single-word typename neither in [value-initialization](https://en.cppreference.com/w/cpp/language/value_initialization) nor in [explicit-type-conversion](https://en.cppreference.com/w/cpp/language/explicit_cast) (applicable point 4).
However in [explicit-type-conversion](https://en.cppreference.com/w/cpp/language/explicit_cast) conversion points 2 and 5 it tells the type must be single-word.